data/service/serialized-data-service.mod/teach/README.md

```diff
@@ -1,76 +1,230 @@
 # SerializedDataService
 
-A Data Service meant to return data objects serialized in Mod serialization file - .mjson
-Those data objects need to have a UUID assigned as their identifier in their serialization 
-to ensure repeated execution is idempotent.
+## 1. Overview
 
-## Generating UUIDs
+The `SerializedDataService` is responsible for reading and serving data objects that are deserialized from `.mjson` (Mod serialization) files.
 
-We use UUID v7. The best way is to use this:
+Its primary function is to provide a consistent read-only interface for static and instance-level data. It must handle data requests (`readOperation`) from clients, retrieve the appropriate serialized data, deserialize it, and return it in the correct format.
 
-const uuid = require("core/uuid");
-uuid.generate(undefined,true) -> '019a245c-1467-73c0-9c0e-c6e76846030f'
+A core requirement is to ensure all read operations are idempotent, which is achieved through mandatory UUIDs for all serialized objects.
 
-And store it manually in the serialization
+## 2. Core Requirements
 
+### 2.1. Data Format
 
-## Serializing Object by Object vs Serializing all together
+All data served by this service **MUST** be stored in the `.mjson` serialization format.
 
-We either serialize all instances in one file per type, or we index/list them in one file and point to one file per instance. Pro/cons on both, to be experimented with. The second route is shown in data/instance/party/role.mod - meant to start that conversation.
+### 2.2. Object Identification & Idempotency
 
-data/instance/party/education/education-experience.mjson shows them all in one file.
+To ensure that repeated deserialization operations are idempotent (i.e., they do not create duplicate objects), every serialized object instance **MUST** have a persistent `UUID v7` assigned as its identifier.
```

Comment:
they do not create duplicate objects if eventually stored into a database via another data service